lnrpc/routerrpc/router_server.go

```diff
@@ -740,10 +744,55 @@ func (s *Server) trackPayment(identifier lntypes.Hash,
 	case err != nil:
 		return err
 	}
+
+	// Stream updates to the client
+	err = s.trackPaymentStream(
+		subscription,
+		noInflightUpdates,
+		stream.Send,
+		stream.Context,
+	)
+
+	if errors.Is(err, context.Canceled) {
+		log.Debugf("Payment stream %v canceled", identifier)
+	}
+
+	return err
+}
+
+// TrackPayments returns a stream of payment state updates.
+func (s *Server) TrackPayments(request *TrackPaymentsRequest,
+	stream Router_TrackPaymentsServer) error {
+
+	log.Debug("TrackPayments called")
+
+	router := s.cfg.RouterBackend
+
+	// Subscribe to payments.
+	subscription, err := router.Tower.SubscribeAllPayments()
+	if err != nil {
+		return err
+	}
+
+	// Stream updates to the client
+	return s.trackPaymentStream(
+		subscription,
+		request.NoInflightUpdates,
+		stream.Send,
+		stream.Context,
+	)
+}
+
+// trackPaymentStream streams payment updates to the client.
+func (s *Server) trackPaymentStream(
+	subscription *routing.ControlTowerSubscriber,
+	noInflightUpdates bool,
+	send func(*lnrpc.Payment) error,
+	context func() context.Context) error {
```

Comment:
The context should remain the same throughout a call, shouldn't it? So instead of passing in a function to get the context, can't the context itself be passed here directly?

Reply:
Done